x-pack/metricbeat/module/azure/azure.go

```diff
@@ -96,9 +96,13 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 	// depending on metric time grain (check `MetricRegistry`
 	// for more information).
 	//
-	// We truncate the reference time to the second to avoid millisecond
-	// variations in the collection period causing skipped collections.
-	referenceTime := time.Now().UTC().Truncate(time.Second)
+	// We round the reference time to the nearest second to avoid
+	// millisecond variations in the collection period causing
+	// skipped collections.
+	//
+	// See "Round outer limits" and "Round inner limits" tests in
+	// the metric_registry_test.go for more information.
+	referenceTime := time.Now().UTC().Round(time.Second)
```

Comment:
to what extent do we care about duplicate collections vs skipped collections? 

i.e. does it make sense to widen the window here to allow for drift in the collection period, maybe up to 5s?

Reply:
Yeah, this jitter I added with https://github.com/elastic/beats/pull/37365/commits/72c5b694dc43010d841c0604f33d560631213b8a allows the collection time to drift a little bit to compensate fluctuations. 

I am currently using a 1-second jitter, but we can go with 2-5s I guess.